extractor/src/main/scala/org/jetbrains/sbt/extractors/DependenciesExtractor.scala

```diff
@@ -95,19 +131,103 @@ object DependenciesExtractor extends SbtStateOps with TaskOps {
         .map(throwExceptionIfUpdateFailed)
         .onlyIf(options.download)
 
+    val allSourceConfigurationsTask = StructureKeys.allSourceConfigurations.in(projectRef).get(state)
+    val allTestConfigurationsTask = StructureKeys.allTestConfigurations.in(projectRef).get(state)
+
+    val classpathConfigurationTask = sbt.Keys.classpathConfiguration.in(projectRef)
+      .forAllConfigurations(state, dependencyConfigurations)
+
+   val settings = Keys.settingsData.value
+
     Def.task {
       (for {
         unmanagedClasspath <- unmanagedClasspathTask
         externalDependencyClasspathOpt <- externalDependencyClasspathTask
+        classpathConfiguration <- classpathConfigurationTask
+        allSourceConfigurations <- allSourceConfigurationsTask
+        allTestConfigurations <- allTestConfigurationsTask
       } yield {
+
+        val projectToTransitiveDependencies = getTransitiveDependenciesForProject(
```

Comment:
Actually, when `options.insertProjectTransitiveDependencies` is `false` I shouldn't perform this operation. The way it is done now, all performance tests are not correct